src/setup_node_env/patches/child_process.js

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+var op = require('object-prototype');
+
+var create = op.create;
+var assign = op.assign;
+
+// Ensure, when spawning a new child process, that the `options` and the
+// `options.env` object passed to the child process function doesn't inherit
+// from `Object.prototype`. This protects against similar RCE vulnerabilities
+// as described in CVE-2019-7609
+module.exports = function(cp) {
+  cp.exec = new Proxy(cp.exec, { apply: patchOptions() });
```

Comment:
The `exec` function actually just calls the `execFile` function internally in Node.js, so we would get the same behavior if we removed this Proxy. In fact, keeping this Proxy means that we'll patch the `options` object twice whenever `exec` is called.

On the other hand, leaving out `exec` from the list might look like a mistake to someone reading the code later (though that can be fixed with a comment) and the fact that `exec` calls `execFile` internally in Node.js is an undocumented implementation detail that might go away without any warning. Though, in that case our tests would catch it.

@kobelb I'm leaning towards not patching `exec` in this case. Do you agree?

Reply:
As we discussed on Zoom, I think that's entirely reasonable as long as our tests will begin to fail if there's a change to Node.js's implementation.

Reply:
Updated 👍